prow/cmd/gerrit/main.go

```diff
@@ -97,7 +184,20 @@ func main() {
 		logrus.WithError(err).Fatal("Error getting kube client.")
 	}
 
-	c, err := adapter.NewController(o.lastSyncFallback, o.cookiefilePath, o.projects, kc, cfg)
+	ctx := context.Background() // TODO(fejta): use something better
+	op, err := io.NewOpener(ctx, o.gcsCredentialsFile)
```

Comment:
and don't fail here if user is not using gcs

Reply:
It doesn't. If you don't specify a credentials file you can either use a `/local/path` or else it will probably fail if you use a `gs://path`

Reply:
ack - then we should validate the flags (ensure if you are using gcs, then `gcsCredentialsFile` must set)

Reply:
Hmm, actually things work as you'd expect:

https://github.com/kubernetes/test-infra/blob/0a2e51dc8afa02adebff7c2a5e899a316e042893/pkg/io/opener.go#L57-L67

Reply:
Ah - `storage.NewClient(ctx, options...)` actually won't return error and will give you an anonymous gcs client? :-)

Reply:
https://github.com/googleapis/google-cloud-go/blob/master/storage/storage.go#L96-L102

Like depending on whether you're on GCP, what environment variables, are defined, etc different stuff will happen.

I'm not 100% sure we want to list all that out here?

Reply:
to me it looks like this io.Opener won't really work with local files :-)

Reply:
nvm, cannot read code: https://github.com/kubernetes/test-infra/blob/0a2e51dc8afa02adebff7c2a5e899a316e042893/pkg/io/opener.go#L102-L108